pipeline/export/nwb.py

```diff
@@ -251,216 +362,306 @@ def datajoint_to_nwb(session_key, raw_ephys=False, raw_video=False):
+        description="Timestamps of the photo-stimulation and the corresponding powers (in mW) being applied",
         data=powers.astype(float),
         timestamps=event_starts.astype(float),
-        control=photo_stim.astype('uint8'), control_description=stim_sites)
+        control=photo_stim.astype("uint8"),
+        control_description=stim_sites,
+    )
     behavioral_event.create_timeseries(
-        name='photostim_stop_times', unit='mW', conversion=1.0,
-        description='Timestamps of the photo-stimulation being switched off',
+        name="photostim_stop_times",
+        unit="mW",
+        conversion=1.0,
+        description="Timestamps of the photo-stimulation being switched off",
         data=np.full_like(event_starts.astype(float), 0),
         timestamps=event_stops.astype(float),
-        control=photo_stim.astype('uint8'), control_description=stim_sites)
+        control=photo_stim.astype("uint8"),
+        control_description=stim_sites,
+    )
 
     # ----- Raw Video Files -----
     if raw_video:
-        tracking_files_info = (tracking_ingest.TrackingIngest.TrackingFile & session_key).fetch(
-            as_dict=True, order_by='tracking_device, trial')
+        tracking_files_info = (
+            tracking_ingest.TrackingIngest.TrackingFile & session_key
+        ).fetch(as_dict=True, order_by="tracking_device, trial")
         for tracking_file_info in tracking_files_info:
-            video_path = tracking_root_data_dir / tracking_file_info.pop('tracking_file')
+            video_path = pathlib.Path(
+                tracking_root_data_dir / tracking_file_info.pop("tracking_file")
+            ).with_suffix(".mp4")
             video_metadata = dict(
                 Behavior=dict(
                     Movies=[
                         dict(
-                            name=video_path,
+                            name=str(video_path),
                             description=video_path.as_posix(),
                             unit="n.a.",
-                            format='external',
+                            format="external",
                             starting_frame=[0, 0, 0],
-                            comments=str(tracking_file_info))
+                            comments=str(tracking_file_info),
+                        )
                     ]
                 )
             )
-            MovieInterface([video_path]).run_conversion(nwbfile=nwbfile, metadata=video_metadata, external_mode=False)
-    
+            try:
+                MovieInterface([video_path]).run_conversion(
+                    nwbfile=nwbfile, metadata=video_metadata, external_mode=False
+                )
+            except FileNotFoundError:
+                warnings.warn(f"No raw video file found at {video_path}.")
+                continue
+
     return nwbfile
 
 
-def export_recording(session_keys, output_dir='./', overwrite=False):
+def export_recording(session_keys, output_dir="./", overwrite=False):
```

Comment:
Ultimately, this can be the function to use for export. It calls the `datajoint_to_nwb` function for every session in the `session_keys` list.